rfcs/0192-fr-copy-receipt-performance.md

```diff
@@ -0,0 +1,218 @@
+- Feature Name: `fr_copy_receipt_performance`
+- Start Date: 2023-08-01
+- RFC PR: [fiskaltrust/middleware#192](https://github.com/fiskaltrust/middleware/pull/192)
+<!-- - Tracking Issue: [fiskaltrust/middleware#0000](https://github.com/fiskaltrust/middleware/issues/0000) -->
+
+# Summary
+
+Adding a `ftJournalFRCopyPayload` table in the FR Queue where we store copy receipts.
+
+# Motivation
+
+The current implementation of the CopyReceipt is performing badly.
+Processing a CopyReceipt takes an increasing amount of time for each previous CopyReceipt in the queue.
+This inevitably leads to a point where the CopyReceipts are not processed in time.
+
+Currently when processing a CopyReceipt we have iterate over all previous CopyReceipts in the `ftJournalFR` table, deserialize the response and check if it is using the same `cbPreviousReceiptReference`.
+We then count all of those CopyReceipts and return that as the count.
+
+# Guide-level explanation
+
+For the PosCreator they are no implementation changes needed, the CopyReceipts still work like they did before.
+
+## New local queues
+
+New Queues don't need to take anything into account.
+They will work out of the box.
+
+## Updating local queues
+
+The new table will be created on start up after updating the queue.
+
+The new `ftJournalFRCopyPayload` table will be populated with the `CopyPayload` (The deserialized JWT) data of each CopyReceipt from the `ftJournalFR` table.
+
+## SignatureCloud
+
+The SignatureCloud will be updated as well.
+The new version will not automatically be used since this would mean a recertification of the PosCreator.
+Instead we will time the switch to the new version accordingly with the PosCreators.
+
+# Reference-level explanation
+
+We will add a migration which adds the new table `ftJournalFRCopyPayload`.
+
+This Table will contain the following columns:
```

Comment:
Might make sense to also add the `CNumerator` here and sort by that DESC, right? By doing this, we'll only have to load a single row and can
a) make the lookup faster in Azure Storage by sorting DESC by this
b) make the lookup faster in SQL databases by putting an index on this field.